apollo-federation/src/sources/connect/validation/expression.rs

```diff
@@ -84,7 +84,25 @@ pub(crate) fn validate(expression: &Expression, context: &Context) -> Result<(),
         expression,
         location,
     } = expression;
-    let shape = expression.shape();
+
+    // let var_shapes: IndexMap<String, Shape> = context
+    //     .var_lookup
+    //     .iter()
+    //     .map(|(name, shape)| (name.to_string(), shape.clone()))
+    //     .collect();
+
+    let shaped_selection = expression.shaped_selection();
+    // Refine shaped_selection with additional named shapes derived from
+    // context.schema, which typically helps eliminate errors and unknown
+    // shapes from the computed output shape.
+    //
+    // TODO Reenable this when we can deal with the consequences better,
+    // using (for example) tools for mapping GraphQL null values to None
+    // when appropriate.
+    // .refine(var_shapes);
+
+    let shape = shaped_selection.output_shape();
+
     let errors: Vec<Error> = shape
         .errors()
         .map(|err| Error {
```

Comment:
Here's where I temporarily commented out the call to `ShapedSelection::refine` that specifies better variable shapes. @dylan-apollo may want to try un-commenting this code to see what fails.